examples/local_damage.py

```diff
@@ -0,0 +1,249 @@
+        if d is None:
+            d = self.d
+        return df.sym(df.grad(d))
+    
+    def sigma(self):
+        eps, E, nu = self.eps(), self.mat.E, self.mat.nu
+        lmbda = E * nu / (1. + nu) / (1. - 2. * nu)
+        mu = E / 2. / (1. + nu)
+
+        return 2 * mu * eps + lmbda * df.tr(eps) * df.Identity(2)
+
+    def traction(self, n):
+        return df.dot((1.-omega(self.k))*self.sigma(), n)
+
+    def kappa(self):
+        return max(self.k, self.eeq(self.eps(self.d)))
+
+    def update(self):
+        self.k.assign(df.project(self.kappa()))
+        
+    def get_solver(self, bcs):
+        df.parameters["form_compiler"]["quadrature_degree"] = 2
+
+        v = df.TestFunction(self.W)
+        F = df.inner(self.eps(v), (1.-omega(self.kappa())) * self.sigma()) * df.dx
+        J = df.derivative(F, self.d)
+
+        problem = df.NonlinearVariationalProblem(F, self.d, bcs, J=J)
+        solver = df.NonlinearVariationalSolver(problem)
+        solver.parameters["nonlinear_solver"] = "snes"
+        solver.parameters["symmetric"] = True
+        solver.parameters["snes_solver"]["error_on_nonconvergence"] = False
+        solver.parameters["snes_solver"]["line_search"] = "bt"
+        solver.parameters["snes_solver"]["linear_solver"] = "mumps"
+        solver.parameters["snes_solver"]["maximum_iterations"] = 10
+        solver.parameters["snes_solver"]["report"] = False
+
+        return solver
+
+
+class LoadDisplacementCurve:
+    def __init__(self, model, boundary, marker=6174, direction=None):
+        """
+        Postprocessing class to track the tractions at `boundary` (actually 
+        tractions . direction) and the displacements at the boundary.
+
+        Each update to the plot can visualized using 
+            fenics_helpers.plotting.AdaptivePlot
+        """
+        mesh = model.mesh
+        self.model = model
+        boundary_markers = df.MeshFunction("size_t", mesh, mesh.topology().dim() - 1)
+
+        boundary.mark(boundary_markers, marker)
+        ds = df.Measure("ds", domain=mesh, subdomain_data=boundary_markers)
+
+        n = df.FacetNormal(mesh)
+        if direction is None:
+            direction = n
+        self.boundary_load = df.dot(model.traction(n), direction) * ds(marker)
+        self.boundary_disp = df.dot(model.d, direction) * ds(marker)
+        self.area = df.assemble(1.0 * ds(marker))
+
+        self.load = []
+        self.disp = []
+
+        self.plot = None
+
+    def __call__(self, t):
+        load = df.assemble(self.boundary_load)
+        disp = df.assemble(self.boundary_disp) / self.area
+        self.load.append(load)
+        self.disp.append(disp)
+        if self.plot is not None:
+            self.plot(disp, load)
+
+    def show(self, fmt="-rx"):
+        self.plot = fh.plotting.AdaptivePlot(fmt)
+
+class Plotter:
+    def __init__(self, model, filename=None):
+        self.model = model
+        if filename is None:
+            filename = "out.xdmf"
```

Comment:
Seems like you could just say `def __init__(self, model, filename="out.xdmf"):`, no?